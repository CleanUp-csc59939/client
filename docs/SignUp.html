<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: SignUp</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: SignUp</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>SignUp<span class="signature">()</span><span class="type-signature"> &rarr; {Component}</span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="SignUp"><span class="type-signature"></span>new SignUp<span class="signature">()</span><span class="type-signature"> &rarr; {Component}</span></h4>
    

    



<div class="description">
    <p>return Sign Up page*</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_signup_SignUp.js.html">components/signup/SignUp.js</a>, <a href="components_signup_SignUp.js.html#line15">line 15</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>Return Home page showing the user's events</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">Component</span>


    </dd>
</dl>

    





    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="~onFinish"><span class="type-signature">(inner) </span>onFinish<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>Calls the create user register service with user inputs</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_signup_SignUp.js.html">components/signup/SignUp.js</a>, <a href="components_signup_SignUp.js.html#line25">line 25</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="Banner.html">Banner</a></li><li><a href="CreateMeetUp.html">CreateMeetUp</a></li><li><a href="EventPics.html">EventPics</a></li><li><a href="Header.html">Header</a></li><li><a href="Home.html">Home</a></li><li><a href="Login.html">Login</a></li><li><a href="MyMeetups.html">MyMeetups</a></li><li><a href="Profile.html">Profile</a></li><li><a href="ProfilePic.html">ProfilePic</a></li><li><a href="ProtectedRoute.html">ProtectedRoute</a></li><li><a href="SignUp.html">SignUp</a></li><li><a href="SingleEvent.html">SingleEvent</a></li></ul><h3>Global</h3><ul><li><a href="global.html#authHeader">authHeader</a></li><li><a href="global.html#ConvertDate">ConvertDate</a></li><li><a href="global.html#CustomSearch">CustomSearch</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.7</a> on Tue May 25 2021 02:22:28 GMT-0400 (Eastern Daylight Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>