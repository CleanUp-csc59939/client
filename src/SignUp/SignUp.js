import React from 'react';
import '../CSS/Header.css';
import { FormWeb, FormMobile } from './Form';

const SignUp = () => {
  const onFinish = (values) => {
    console.log('Success:', values);
    fetch('http://localhost:8080/api/user/register', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(values),
    });
  };

  const onFinishFailed = (errorInfo) => {
    console.log('Failed:', errorInfo);
  };

  return (
    <div>
      <div className='web'>
        <FormWeb onFinish={onFinish} onFinishFailed={onFinishFailed} />
      </div>
      <div className='mobile'>
        <FormMobile onFinish={onFinish} onFinishFailed={onFinishFailed} />
      </div>
    </div>
  );
};

export default SignUp;
