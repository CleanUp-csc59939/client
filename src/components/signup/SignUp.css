@import url(https://fonts.googleapis.com/css?family=Roboto:300);

.login-page {
  width: 650px;
  padding: 8% 0 0;
  margin: auto;
}
.alert {
  margin-bottom: 30px;

  background-color: rgba(248, 215, 218, 1);
  border-color: rgba(220, 53, 69, 1);
  color: rgba(114, 28, 36, 1);
}
.signUpForm {
  position: relative;
  z-index: 1;
  background: #ffffff;
  min-width: 650px;
  margin: 0 0 0px;
  padding: 45px;
  text-align: center;
  box-shadow: 0 0 20px 0 rgba(0, 0, 0, 0.2), 0 5px 5px 0 rgba(0, 0, 0, 0.24);
}



.signUpForm button{
    font-family: 'Roboto', sans-serif;
    text-transform: uppercase;
    outline: 0;
    background: #4caf50;
    width: 100%;
    border: 0;
    color: #ffffff;
    font-size: 14px;
    -webkit-transition: all 0.3 ease;
    transition: all 0.3 ease;
    cursor: pointer;
}
.signUpForm button:hover,
.signUpForm button:active,
.signUpForm button:focus {
  background: #43a047;
}
.signUpForm .message {
  margin: 15px 0 0;
  color: #b3b3b3;
  font-size: 12px;
}
.signUpForm .message a {
  color: #4caf50;
  text-decoration: none;
}
.signUpForm .register-signUpForm {
  display: none;
}
.container {
  position: relative;
  z-index: 1;
  width: 300px;
  margin: 0 auto;
}
.container:before,
.container:after {
  content: '';
  display: block;
  clear: both;
}
.container .info {
  margin: 50px auto;
  text-align: center;
}
.container .info h1 {
  margin: 0 0 15px;
  padding: 0;
  font-size: 36px;
  font-weight: 300;
  color: #1a1a1a;
}
.container .info span {
  color: #4d4d4d;
  font-size: 12px;
}
.container .info span a {
  color: #000000;
  text-decoration: none;
}
.container .info span .fa {
  color: #ef3b3a;
}
.loginBody {
  background: #76b852; /* fallback for old browsers */
  background: -webkit-linear-gradient(right, #76b852, #8dc26f);
  background: -moz-linear-gradient(right, #76b852, #8dc26f);
  background: -o-linear-gradient(right, #76b852, #8dc26f);
  background: linear-gradient(to left, #76b852, #8dc26f);
  font-family: 'Roboto', sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  height: 100vh;
  width: 100vw;
}

.loader {
  border: 16px solid #f3f3f3; /* Light grey */
  border-top: 16px solid #3498db; /* Blue */
  border-radius: 50%;
  width: 15px;
  height: 15px;
  animation: spin 2s linear infinite;
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
