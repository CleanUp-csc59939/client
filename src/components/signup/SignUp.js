import React from 'react';
import '../../Shared/Header.css';
import { FormWeb, FormMobile } from './Form';

const SignUp = () => {
  const onFinish = (values) => {
    console.log('Success:', values);
    fetch('http://localhost:8080/api/user/register', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(values),
    });
  };

  const onFinishFailed = (errorInfo) => {
    console.log('Failed:', errorInfo);
  };

  return (
    <div>
      <h1 style={{ textAlign: 'center' }}>Register</h1>
      <div className='web'>
        <FormWeb onFinish={onFinish} onFinishFailed={onFinishFailed} />
      </div>
      <div className='mobile'>
        <FormMobile onFinish={onFinish} onFinishFailed={onFinishFailed} />
      </div>
    </div>
  );
};

export default SignUp;
