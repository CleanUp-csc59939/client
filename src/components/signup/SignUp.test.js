import React from 'react';
import { shallow } from 'enzyme';
import SignUp from './SignUp';

describe('SignUp', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<SignUp debug />);

    expect(component).toMatchSnapshot();
  });
});
