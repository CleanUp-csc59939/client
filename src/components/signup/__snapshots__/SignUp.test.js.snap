// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should render correctly in "debug" mode 1`] = `
<div>
  <div
    className="web"
  >
    <div
      className="loginBody"
    >
      <div
        className="login-page"
      >
        <h1
          style={
            Object {
              "color": "whitesmoke",
              "fontSize": "36px",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Register
        </h1>
        <div
          className="signUpForm"
        >
          <FormWeb
            loading={false}
            onFinish={[Function]}
            onFinishFailed={[Function]}
            setLoading={[Function]}
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="mobile"
  >
    <FormMobile
      onFinish={[Function]}
      onFinishFailed={[Function]}
    />
  </div>
</div>
`;
